skbio/tree/_tree.py

```diff
@@ -617,6 +617,77 @@ def lowest_common_ancestor(self, tipnames):
 
     lca = lowest_common_ancestor  # for convenience
 
+    def path(self, other, include_ends=False):
+        r"""Return the list of nodes in the path from one node to another.
+
+        Parameters
+        ----------
+        other : TreeNode
+            Final node of path.
+        include_ends: bool, optional
+            Whether to include the initial and final nodes in the list.
+
+        Returns
+        -------
+        list
+            List of TreeNode objects.
+
+        Examples
+        --------
+        >>> from skbio import TreeNode
+        >>> tree = TreeNode.read(["((a,b)c,(d,e)f)root;"])
+        >>> node_1, node_2 = tree.find('a'), tree.find('d')
+        >>> path = node_1.path(node_2)
+        >>> print(len(path))
+        3
+        >>> path_2 = node_1.path(node_2, include_ends=True)
+        >>> print(len(path_2))
+        5
+
+        """
+        # check to see if nodes are on same tree
+        root1, root2 = self.root(), other.root()
+        if root1 is not root2:
+            raise TypeError("Could not find path between nodes.")
+
+        # create list of ancestors for both nodes
+        anc1, anc2 = (
+            [node for node in self.ancestors()],
+            [node for node in other.ancestors()],
+        )
+
+        # find lowest common ancestor
+        for n1, n2 in zip(reversed(anc1), reversed(anc2)):
+            if n1 is n2:
+                lca = n1
+            else:
+                break
+
+        # delete nodes from intersection of ancestors
+        del anc1[anc1.index(lca) :]
+        del anc2[anc2.index(lca) :]
+
+        # reverse the second list for correct order
+        anc2.reverse()
+
+        # handle cases for nodes that are ancestors of each other
```

Comment:
I am not sure if the instances of nodes being ancestors of each other need to be singled out. What if we don't?

Reply:
The path between a node and an ancestor will be the complement of the intersection of the two ancestor lists. Otherwise, the path will be the union of the two ancestor lists and subtracting their intersection while keeping the lowest common ancestor. 

These two constructions are not equivalent. For example, the lowest common ancestor is not in the path between a node and an ancestor, but does appear in the path otherwise.

Reply:
My understanding is that if node A is ancestral to node B, then the LCA of A and B is A, and the path from A to B is A straight down to B. So LCA is indeed in the path, as long as `include_ends=True`. I think the logic of your code below is correct. But I wonder even without the two `if`s above, it will still work. I am not sure if my guess is correct (because I didn't code). Maybe you can try?

Reply:
Your idea is correct with my code as I add the nodes themselves to the beginning of their ancestor lists in lines 650-651, since the ancestor function does not include the node itself in the list. 

However, maybe I should note that the lowest common ancestor function in TreeNode does not have this behavior, and will return the parent of A rather than A in your example.

Reply:
You are right. I tested your code and mine and found that your code works.